libcloud/__init__.py

```diff
@@ -75,3 +94,40 @@ def _init_once():
             paramiko.common.logging.basicConfig(level=paramiko.common.DEBUG)
 
 _init_once()
+
+
+class DriverType:
+    """ Backup-as-a-service driver """
+    BACKUP = BackupProvider
+
+    """ Compute-as-a-Service driver """
+    COMPUTE = ComputeProvider
+
+    """ Container-as-a-Service driver """
+    CONTAINER = ContainerProvider
+
+    """ DNS service provider driver """
+    DNS = DnsProvider
+
+    """ Load balancer provider-driver """
+    LOADBALANCER = LoadBalancerProvider
+
+    """ Storage-as-a-Service driver """
+    STORAGE = StorageProvider
+
+
+DriverTypeFactoryMap = {
+    DriverType.BACKUP: get_backup_driver,
+    DriverType.COMPUTE: get_compute_driver,
+    DriverType.CONTAINER: get_container_driver,
+    DriverType.DNS: get_dns_driver,
+    DriverType.LOADBALANCER: get_loadbalancer_driver,
+    DriverType.STORAGE: get_storage_driver
+}
+
+
+def get_driver(type, provider):
+    """
+    Get a driver
+    """
+    return DriverTypeFactoryMap[type](provider)
```

Comment:
We should probably re-throw a more user-friendly exception in case API / driver is not found, right?

Since I imagine KeyError or similar might be a bit confusing and not immediately obvious to the user what is going on.